### PR TITLE
Remove meta.edn

### PR DESCRIPTION
**Pull request by podviaznikov**
All metadata stored now as clojure metadata associated with boot fileset.

### CODE REVIEW

**Comment by martinklepsch** [comment]
cc @Deraen

**Comment by podviaznikov** [comment]
@Deraen I removed `meta.edn`. It's one of the changes you suggested in #2. Structure of metadata is the same for now. I'll probably do another PR for that.

Let me know if you have comments about this PR.

**Comment by podviaznikov** [comment]
Reminder to myself: still need to update reamde.

File: src/io/perun.clj [hunk 1]
```diff
@@ -36,50 +36,48 @@
...
-        (commit fileset tmp)))))
+      (let [markdown-files (->> fileset boot/user-files (boot/by-ext ["md" "markdown"]) (map #(.getPath (boot/tmp-file %))))
+            parsed-metadata (pod/with-call-in @pod
+                              (io.perun.markdown/parse-markdown
+                                ~(.getPath tmp)
+                                ~options
+                                ~markdown-files))
+            fs-with-meta (with-meta fileset {:metadata parsed-metadata})]
+        (u/info "Parsed markdown files\n")
+        (commit fs-with-meta tmp)))))
```
Comment (Deraen) [inline src/io/perun.clj]: Markdown task is not writing to any files? tmp-dir and commit can be removed.

**Comment by Deraen** [comment]
Other than my comment about markdown task, looks good!

File: src/io/perun.clj [hunk 2]
```diff
@@ -36,50 +36,48 @@
 
 (deftask markdown
   "Parse markdown files"
-  [d datafile        DATAFILE        str "Target datafile with all parsed meta information"
-   f create-filename CREATE_FILENAME str "Function that creates final target filename of the file"]
+  [f create-filename CREATE_FILENAME str "Function that creates final target filename of the file"]
```
Comment (martinklepsch) [inline src/io/perun.clj:39]: not really related to this PR but `create-filename` does not seem to be used anywhere?

File: src/io/perun.clj (hunk 1 above)
Comment (martinklepsch) [inline src/io/perun.clj]: but it adds metadata — does that not require `commit!`?

File: src/io/perun.clj (hunk 2 above)
Comment (Deraen) [inline src/io/perun.clj:39]: It's used in markdown/process-file

Comment (podviaznikov) [inline src/io/perun.clj:39]: It's used here: https://github.com/hashobject/perun/blob/remove-meta-edn/src/io/perun/markdown.clj#L55. Passed as option.

File: src/io/perun.clj (hunk 1 above)
Comment (Deraen) [inline src/io/perun.clj]: Nope. Commit is to write changes to disk. Clojure metadata is just associated with Java object.

Comment (podviaznikov) [inline src/io/perun.clj]: Yeah, just double checked. `commit!` is not required. Thanks for the catch

**Comment by podviaznikov** [comment]
Thanks for the review @Deraen and @martinklepsch

**Pull request merged**